### ISSUE TITLE
mongodb connection loss crashes the app

### ISSUE THREAD

**Issue by kristofsajdak** [agco/harvesterjs#133]
when a harvesterjs app is running, and the connection to Mongodb is lost at some point, it will crash the application with the following error message : 

events.js:72
        throw er; // Unhandled 'error' event
              ^
Error: failed to connect to [192.168.59.103:27017]
    at null.<anonymous> (/Users/CDBKRS/projects/harvesterjs/node_modules/mongoose/node_modules/mongodb/lib/mongodb/connection/server.js:555:74)
    at emit (events.js:106:17)
    at null.<anonymous> (/Users/CDBKRS/projects/harvesterjs/node_modules/mongoose/node_modules/mongodb/lib/mongodb/connection/connection_pool.js:156:15)
    at emit (events.js:98:17)
    at Socket.<anonymous> (/Users/CDBKRS/projects/harvesterjs/node_modules/mongoose/node_modules/mongodb/lib/mongodb/connection/connection.js:534:10)
    at Socket.emit (events.js:95:17)
    at net.js:440:14
    at process._tickCallback (node.js:419:1

**Comment by kristofsajdak** [comment]
Apparently this is a Mongoose level issue, the version which we are using doesn't come with a default resilience mechanism. 

Applying what I found as an accepted answer on Stackoverflow : http://stackoverflow.com/questions/10873199/how-to-handle-mongoose-db-connection-interruptions

It feels like a hack, but should do for now. Did some tests where I stopped and started Mongodb, and it does reconnect automatically.
